toolkit/scripts/check_required_kernel_configs.py

```diff
@@ -0,0 +1,123 @@
+            else:
+                print_data.append([config_option, value['arch'], value['value'], "OK"])
+    # Calculate maximum width for each column
+    column_widths = [max(len(str(row[i])) for row in print_data) for i in range(len(print_data[0]))]
+
+    # Print columns
+    for j, row in enumerate(print_data):
+        for i, column in enumerate(row):
+            print(str(column).ljust(column_widths[i] + 2), end='')
+        print()
+        if j == 0:
+            print("-------------------------------------------------------------------------------")
+
+
+## Main
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="Tool for checking if known required kernel configs are present.")
+
+
+    parser.add_argument('--required_configs', help='path to json of required configs', required=True)
+    parser.add_argument('--config_file', help='path to config being checked', required=True)
+    parser.add_argument('--verbose', action='store_true', help='get full report', required=False)
+    args = parser.parse_args()
+    required_configs = args.required_configs
+    config_file = args.config_file
+
+    kernel = extract_kernel_dir_name(config_file)
+    if kernel == None:
+        print("Kernel not found in config filepath")
+        sys.exit(1)
+
+    input_config_data = get_data_from_config(config_file)
+
+    arch = extract_config_arch(input_config_data)
+    if arch == None:
+        print("Architecture not found in config file")
+        sys.exit(1)
+
+    config_map = create_map_of_config_values(input_config_data)
+
+    print()
+    print("===============================================================================")
+    print(f"== Results for {config_file} ==")
+    print("===============================================================================")
+
+    # result is map: {config: (newValue, expectedValue, comment, PR)}
+    result = check_required_configs_in_configfile(required_configs, kernel, arch, config_map)
+    # check if required configs are present
+    # not an error is not all kernels are being checked
+    if result == None:
+        print(f"No required configs for {kernel} in json")
+        sys.exit(0)
+    
+    if args.verbose:
+        print_verbose(required_configs, kernel, arch, result)
+    else:
+        if result == {}:
+            print("All required configs are present")
+        else:
+            print()
+            print ("----------------- Kernel config verification FAILED -----------------")
+            for config_option, value in result.items():
+                print(f'{config_option} is "{result[config_option]["newValue"]}", expected {result[config_option]["expectedValue"]}.\nReason: {result[config_option]["comment"]}')
+                print(f'PR: {result[config_option]["PR"]}')
+                print()
```

Comment:
One more thing I remembered: if you want to do multi-line prints, you can also do this to avoid multiple `print` calls, using `\n`, and make the code reflect the console output a bit closer:
```suggestion
                print(f'''
{config_option} is "{result[config_option]["newValue"]}", expected {result[config_option]["expectedValue"]}.
Reason: {result[config_option]["comment"]}')
PR: {result[config_option]["PR"]}'

```